### ISSUE TITLE
Allow Complex (e.g., JSON) Default Property Values

### ISSUE THREAD

**Issue by Nicholas-Westby** [kgiszewski/Archetype#417]
When adding a property to an Archetype fieldset, a default value can be specified:

![complex-default](https://user-images.githubusercontent.com/5933222/28188793-3bebc8a0-67d8-11e7-8ec5-8c76316e5a66.png)

That works OK when the value is a simple text value. However, it doesn't seem to work when it's a complex data type (in this case, a blob of JSON).

On a related note, would be great if this were a text area rather than a text line (that way, complex data that spans multiple lines can more easily be read).

FYI, the data type I'm trying to set a default for in this case is a Formulate form configuration picker.

**Comment by Nicholas-Westby** [comment]
By the way, here's how Umbraco correctly stores a picked form value:

```javascript
{"alias":"form","value":"{\r\n  \"id\": \"397c9fbd642a4e2098487572fafe7338\"\r\n}"}
```

My guess would be that the way this is stored is not how it's sent to the AngularJS directive. I would guess it converts the string value into a JSON object. I would further suspect that the reason this doesn't work with Archetype is that Archetype is setting this on the client side, which bypasses Umbraco's server-side JSON detection logic (i.e., it is passed as a string rather than as JSON). Haven't actually looked at the code though, so that's just a guess.

The simplest solution I can think of would be to add a checkbox under the "Default Value" field that says "Treat Default Value as JSON?" Or maybe you could have a drop down that indicates the format of the default value (the initial value being "Plain Text", and another option being "JSON").

**Comment by kjac** [comment]
Actually Archetype already tries to parse the default values to JSON objects. Unfortunately there was an error in the subsequent value assignment, which forced the parsed object into a string.

It's fixed, will be part of the next release.

Please note that the JSON used for default values *must* validate with https://jsonlint.com/ - or in other words, must be valid for `JSON.parse()`.

**Comment by kjac** [comment]
Reopening this until we have a new release.

**Comment by kjac** [comment]
Fixed in v1.15.1
